autode/wrappers/G09.py

```diff
@@ -36,180 +43,261 @@ def modify_keywords_for_point_charges(keywords):
-                cart_calc.output_filename = None
-                cart_calc.output_file_exists = False
-                cart_calc.terminated_normally = False
-                cart_calc.output_file_lines = None
-                cart_calc.rev_output_file_lines = None
-                cart_calc.run()
-                for line in cart_calc.rev_output_file_lines:
-                    if 'Normal termination of Gaussian' in line:
-                        calc.name = cart_calc.name
-                        calc.output_filename = cart_calc.output_filename
-                        calc.set_output_file_lines()
-                        logger.info('The cartesian optimisation converged')
-                        return True
-
-                logger.info('Returning to internal coordinates')
-
-                fixed_angle_calc = deepcopy(calc)
-                fixed_angle_calc.name += '_internal'
-                fixed_angle_calc.xyzs = cart_calc.get_final_atoms()
-                fixed_angle_calc.input_filename = None
-                fixed_angle_calc.output_filename = None
-                fixed_angle_calc.output_file_exists = False
-                fixed_angle_calc.terminated_normally = False
-                fixed_angle_calc.output_file_lines = None
-                fixed_angle_calc.rev_output_file_lines = None
-                fixed_angle_calc.run()
-                if fixed_angle_calc.terminated_normally:
-                    logger.info('The 180° angle issue has been fixed')
-                    calc.output_filename = fixed_angle_calc.output_filename
-                    calc.name = fixed_angle_calc.name
-                    calc.set_output_file_lines()
-                    return True
-                else:
-                    return False
+                logger.warning('Gaussian encountered a 180° angle and crashed')
+                break
+
+        if not rerun_if_failed:
+            return False
+
+        # Set a limit on the amount of times we do this
+        if calc.name.endswith('internal_internal_internal_internal'):
+            return False
+
+        fixed_calc = rerun_angle_failure(calc)
+
+        if fixed_calc.terminated_normally():
+            logger.info('The 180° angle issue has been fixed')
+            calc.output = fixed_calc.output
+            calc.name = fixed_calc.name
+            calc.output.set_lines()
+            return True
 
         return False
 
     def get_enthalpy(self, calc):
         """Get the enthalpy (H) from an g09 calculation output"""
 
-        for line in calc.rev_output_file_lines:
+        for line in reversed(calc.output.file_lines):
             if 'Sum of electronic and thermal Enthalpies' in line:
                 return float(line.split()[-1])
 
-        logger.error('Could not get the enthalpy from the calculation. Was a frequency requested?')
+        logger.error('Could not get the enthalpy from the calculation. '
+                     'Was a frequency requested?')
```

Comment:
Maybe this should say "A frequency must be requested"

Reply:
Less passive aggressive ;)